processor/src/main/resources/org/mapstruct/ap/internal/model/StreamMappingMethod.ftl

```diff
@@ -60,7 +60,7 @@
     <#elseif resultType.iterableType>
         <#if existingInstanceMapping>
             ${resultName}.clear();
-        <#elseif needVarDefine>
+        <#elseif needVarDefine || afterMappingReferences?has_content>
```

Comment:
I rather did have this as part of the `needVarDefine` definition (with an OR on `beforeMappingWithTargetReferences`).. but then stuff starts to breakdown..